client/state/user-settings/reducer.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { merge, omit } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	USER_SETTINGS_UPDATE,
+	USER_SETTINGS_UNSAVED_CLEAR,
+	USER_SETTINGS_UNSAVED_SET,
+	USER_SETTINGS_UNSAVED_REMOVE,
+} from 'state/action-types';
+
+export const settings = ( state = false, action ) => {
+	switch ( action.type ) {
+		case USER_SETTINGS_UPDATE:
+			return merge( {}, state, action.settingValues );
+
+		default:
+			return state;
+	}
+};
```

Comment:
Good point, I'm changing the default state value to `null`. The legacy `lib/user-settings` module used `false` and we have a good opportunity to make the change here.

I'd rather keep the reducer as a `switch` statement. It might be a bit more verbose, but it's 100% clear at first glance what the reducer does and which actions it handles. On the other hand, decoding the ternary operator takes some mental effort, at least for me :)